File: app/src/main/java/com/yijian/staff/mvp/huifang/student/task/HuiFangTaskActivity.java
package com.yijian.staff.mvp.huifang.student.task;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.view.View;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.yijian.staff.R;
import com.yijian.staff.bean.HuiFangTypeBean;
import com.yijian.staff.db.DBManager;
import com.yijian.staff.mvp.base.mvc.MvcBaseActivity;
import com.yijian.staff.mvp.huifang.student.history.HuiFangHistoryActivity;
import com.yijian.staff.mvp.huifang.student.task.fragment.BaseHuiFangTaskFragment;
import com.yijian.staff.mvp.huifang.student.task.pageadapter.HuiFangPagerAdapter;
import com.yijian.staff.net.httpmanager.HttpManager;
import com.yijian.staff.bean.HuiFangTypeRequestBody;
import com.yijian.staff.net.response.ResultJSONArrayObserver;
import com.yijian.staff.util.JsonUtil;
import com.yijian.staff.widget.NavigationBar2;
import com.yijian.staff.widget.PagerSlidingTabStrip;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.OnClick;

@Route(path = "/test/5")
public class HuiFangTaskActivity extends MvcBaseActivity {

    @BindView(R.id.tabs)
    PagerSlidingTabStrip tabs;
    @BindView(R.id.viewPager)
    ViewPager viewPager;

    private ArrayList<HuiFangTypeBean> huiFangTypeBeanArrayList = new ArrayList<>();
    private int totalNum;


    @Override
    protected int getLayoutID() {
        return R.layout.activity_hui_fang_task;
    }

    @Override
    protected void initView(Bundle savedInstanceState) {
        initView();
    }

    private void initView() {
        initNavigation();
        initData();
    }


    private void initNavigation() {
        NavigationBar2 navigationBar2 = (NavigationBar2) findViewById(R.id.hui_fang_task_navigation_bar);
        navigationBar2.setTitle("会员回访");
        navigationBar2.hideLeftSecondIv();
        navigationBar2.setBackClickListener(this);
    }

    private void initData() {
        showLoading();
        HuiFangTypeRequestBody body=new HuiFangTypeRequestBody();
        body.setChief(true);
        body.setType(1);
        HttpManager.postHuiFangType(body, new ResultJSONArrayObserver(getLifecycle()) {
            @Override
            public void onSuccess(JSONArray result) {
                hideLoading();
                try {
                    if (result != null) {
                        for (int i = 0; i < result.length(); i++) {
                            JSONObject jsonObject = result.getJSONObject(i);
                            if (jsonObject.has("totalNum")) {
                                totalNum = JsonUtil.getInt(jsonObject, "totalNum");
                            }
                            HuiFangTypeBean huiFangTypeBean = new HuiFangTypeBean(jsonObject);
                            huiFangTypeBeanArrayList.add(huiFangTypeBean);
                        }
                        DBManager.getInstance().insertOrReplaceHuiFangTypeBeans(huiFangTypeBeanArrayList);
                        initIndicatorAndViewPager();
                    }
                } catch (JSONException e) {
                    e.printStackTrace();
                }
            }

            @Override
            public void onFail(String msg) {
                hideLoading();
                showToast(msg);
            }
        });
    }


    private void initIndicatorAndViewPager() {
        List<String> mTitleList = new ArrayList<>();
        List<Fragment> fragmentList = new ArrayList<>();
        for (int i = 0; i < huiFangTypeBeanArrayList.size(); i++) {
            HuiFangTypeBean huiFangTypeBean = huiFangTypeBeanArrayList.get(i);
            mTitleList.add(huiFangTypeBean.getName());
            fragmentList.add(new BaseHuiFangTaskFragment(this, huiFangTypeBean.getMenu()));
        }
        HuiFangPagerAdapter huiFangPagerAdapter = new HuiFangPagerAdapter(getSupportFragmentManager(), fragmentList, mTitleList);
        viewPager.setAdapter(huiFangPagerAdapter);
        tabs.setViewPager(viewPager);
        tabs.updateBubbleNum(0, totalNum);
        //初始化显示第一页
        viewPager.setCurrentItem(0);
    }


    @OnClick({R.id.ll_hui_fang_ji_lu})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.ll_hui_fang_ji_lu:
                startActivity(new Intent(HuiFangTaskActivity.this, HuiFangHistoryActivity.class));
                break;
        }
    }

}

File: club_module/src/main/java/com/yijian/clubmodule/ui/huifang/vip/task/HuiFangTaskActivity.kt
package com.yijian.clubmodule.ui.huifang.vip.task

import android.content.Intent
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.View

import com.alibaba.android.arouter.facade.annotation.Route
import com.yijian.clubmodule.R
import com.yijian.clubmodule.db.bean.HuiFangTypeBean
import com.yijian.commonlib.base.mvc.MvcBaseActivity
import com.yijian.clubmodule.ui.huifang.vip.history.HuiFangHistoryActivity
import com.yijian.clubmodule.ui.huifang.vip.task.fragment.BaseHuiFangTaskFragment
import com.yijian.clubmodule.ui.huifang.vip.task.pageadapter.HuiFangPagerAdapter
import com.yijian.clubmodule.net.httpmanager.HttpManager
import com.yijian.clubmodule.net.requestbody.HuiFangTypeRequestBody
import com.yijian.commonlib.net.response.ResultJSONArrayObserver
import com.yijian.commonlib.widget.NavigationBar
import com.yijian.clubmodule.db.ClubDBManager
import com.yijian.commonlib.util.JsonUtil

import org.json.JSONArray
import org.json.JSONException

import java.util.ArrayList

import kotlinx.android.synthetic.main.activity_hui_fang_task.*

@Route(path = "/test/33")
class HuiFangTaskActivity : MvcBaseActivity() {



    private val huiFangTypeBeanArrayList = ArrayList<HuiFangTypeBean>()
    private var totalNum: Int = 0


    override fun getLayoutID(): Int {
        return R.layout.activity_hui_fang_task
    }

    override fun initView(savedInstanceState: Bundle?) {
        initView()
    }

    private fun initView() {
        initNavigation()
        initData()
        ll_hui_fang_ji_lu.setOnClickListener { _ -> startActivity(Intent(this@HuiFangTaskActivity, HuiFangHistoryActivity::class.java)) }
    }


    private fun initNavigation() {
        val navigationBar = findViewById<View>(R.id.hui_fang_task_navigation_bar) as NavigationBar
        navigationBar.setTitle("会员回访")
        navigationBar.hideLeftSecondIv()
        navigationBar.setBackClickListener(this)
    }


    private fun initData() {
        showLoading()
        val body = HuiFangTypeRequestBody()
        body.isChief = true
        body.type = 0
        HttpManager.postHuiFangType(body, object : ResultJSONArrayObserver(lifecycle) {
            override fun onSuccess(result: JSONArray?) {
                hideLoading()
                try {
                    if (result != null && result.length() > 0) {
                        for (i in 0 until result.length()) {
                            val jsonObject = result.getJSONObject(i)
                            val huiFangTypeBean = HuiFangTypeBean(jsonObject)
                            huiFangTypeBeanArrayList.add(huiFangTypeBean)
                        }
                        val jsonObject = result.getJSONObject(0)
                        if (jsonObject.has("totalNum")) {
                            totalNum = JsonUtil.getInt(jsonObject, "totalNum")
                        }
                        ClubDBManager.getInstance().insertOrReplaceHuiFangTypeBeans(huiFangTypeBeanArrayList)
                        initIndicatorAndViewPager()
                    }
                } catch (e: JSONException) {
                    e.printStackTrace()
                }

            }

            override fun onFail(msg: String) {
                hideLoading()
                showToast(msg)
            }
        })
    }

    private fun initIndicatorAndViewPager() {
        val mTitleList = ArrayList<String>()
        val fragmentList = ArrayList<Fragment>()
        for (i in huiFangTypeBeanArrayList.indices) {
            val huiFangTypeBean = huiFangTypeBeanArrayList[i]
            mTitleList.add(huiFangTypeBean.name)
            fragmentList.add(BaseHuiFangTaskFragment( huiFangTypeBean.menu))
        }
        val huiFangPagerAdapter = HuiFangPagerAdapter(supportFragmentManager, fragmentList, mTitleList)
        viewPager.adapter = huiFangPagerAdapter
        tabs.setViewPager(viewPager)
        tabs.updateBubbleNum(0, totalNum)
        //初始化显示第一页
        viewPager.currentItem = 0
    }



}


